vm-deployment/vm.bicep

```
@@ -0,0 +1,282 @@
+/*
+Virtual machine deployment.
+*/
+
+// Parameters for resource names
+param virtualMachines string = 'iagora'
+param publicIPAddresses string = 'iagora-ip'
+param virtualNetworks string = 'iagora-vnet'
+param networkInterfaces string = 'iagora731_z1'
+param networkSecurityGroups string = 'iagora-nsg'
+
+// SSH public key parameter
+@description('The SSH public key to use for authentication.')
+param sshPublicKey string = 'ssh-rsa AAAAB3...your-public-key... user@domain'
+
+// VM size parameter
+@description('The size of the virtual machine.')
+param vmSize string = 'Standard_B2s'
+
+// Location parameter
+@description('The location where resources are deployed.')
+param location string = resourceGroup().location
+
+// IP address and network parameters
+@description('The static public IP address for the virtual machine.')
+param publicIPAddress string = '4.233.216.162'
+
+@description('The private IP address for the virtual machine.')
+param privateIPAddress string = '10.0.0.4'
+
+@description('The address prefix for the virtual network.')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('The address prefix for the subnet.')
+param subnetAddressPrefix string = '10.0.0.0/24'
+
+// Security rules parameters
+@description('Security rules for the network security group.')
+param securityRules array = [
+  {
+    name: 'SSH'
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '22'
+    sourceAddressPrefix: '*'
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 300
+    direction: 'Inbound'
+  }
+  {
+    name: 'HTTP'
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '80'
+    sourceAddressPrefix: '*'
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 320
+    direction: 'Inbound'
+  }
+  {
+    name: 'HTTPS'
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '443'
+    sourceAddressPrefix: '*'
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 340
+    direction: 'Inbound'
+  }
+]
+
+// Networking security group.
+resource networkSecurityGroups_resource 'Microsoft.Network/networkSecurityGroups@2023-11-01' = {
```

```suggestion
resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2023-11-01' = {
```
Why do we need the `_resource` suffix? Can we use exclusively camelCase in all resource references?